hail/src/main/scala/is/hail/annotations/RegionMemory.scala

```diff
@@ -132,20 +132,22 @@ final class RegionMemory(pool: RegionPool) extends AutoCloseable {
 
   private def releaseNDArrays(): Unit = {
     var i = 0
+    val ndArraysToFree = new LongArrayBuilder()
```

Comment:
Like Tim said, we should try not allocating an object here every time. Let's just make a `freeChunk` method that takes one long, and change `freeChunks` to just call `freeChunk` a bunch of times or something